src/linux/processors.c

```diff
@@ -404,3 +464,66 @@ bool cpuinfo_linux_detect_thread_siblings(
 	}
 }
 
+bool cpuinfo_linux_detect_cluster_cpus(
+	uint32_t max_processors_count,
+	uint32_t processor,
+	cpuinfo_siblings_callback callback,
+	void* context)
+{
+	char cluster_cpus_filename[CLUSTER_CPUS_FILENAME_SIZE];
+	const int chars_formatted = snprintf(
+		cluster_cpus_filename, CLUSTER_CPUS_FILENAME_SIZE, CLUSTER_CPUS_FILENAME_FORMAT, processor);
+	if ((unsigned int) chars_formatted >= CLUSTER_CPUS_FILENAME_SIZE) {
+		cpuinfo_log_warning("failed to format filename for cluster cpus of processor %"PRIu32, processor);
+		return false;
+	}
+
+	struct siblings_context siblings_context = {
+		.group_name = "cluster",
+		.max_processors_count = max_processors_count,
+		.processor = processor,
+		.callback = callback,
+		.callback_context = context,
+	};
+	if (cpuinfo_linux_parse_cpulist(cluster_cpus_filename,
+		(cpuinfo_cpulist_callback) siblings_parser, &siblings_context))
+	{
```

Comment:
Curly bracket on the same line...
```suggestion
		(cpuinfo_cpulist_callback) siblings_parser, &siblings_context)) {
```

Reply:
Fixed, here and elsewhere.